### ISSUE TITLE
Pretzel does not handle :slug setting for permalinks

### ISSUE THREAD

**Issue by stanac** [Code52/pretzel#316]
Problem is that Pretzel doesn't handle `:slug` setting for permalinks.

Steps to reproduce:
1. Create new site with `pretzel create` (I am using razor)
2. Set slug in `permalink` in `_config.yml` so config file looks like this:
```yml
pretzel: 
    engine: razor
permalink: /blog/:year-:month-:day/:slug
```
3. run `bake` command

Problem is that pretzel is trying to create a directory that looks like this for file that looks like this:
`c:\mysite\_site\blog\2017-01-05\:slug\index.html` and it fails with  `System.NotSupportedException`: `The given path's format is not supported.`. Issue is in class `JekyllEngineBase` method `CreateOutputDirectory`

This is from Jekyll docs: https://jekyllrb.com/docs/permalinks/

> Slugified title from the document’s filename (any character except numbers and letters is replaced as hyphen). May be overridden via the document’s slug YAML front matter.

After further investigation i thought that maybe `permalink` should be indented so it falls under `pretzel` node in config file, but it lead to permalink setting being ignored.

Basically it should be easy to fix, I could submit a pull request if you agree.

**Comment by laedit** [comment]
Thanks a lot for your interest in Pretzel ! :smile: 
Indeed it seems that the property `slug` is missing in Pretzel, a PR is welcome.

**Comment by laedit** [comment]
Fixed by #317

**Comment by stanac** [comment]
yes, it looks like someone was faster to fix it :)

**Comment by thoemmi** [comment]
@stanac Sorry, couldn't resist 😉
